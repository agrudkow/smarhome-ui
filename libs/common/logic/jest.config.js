module.exports = {
  name: 'common-logic',
  preset: '../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory: '../../../coverage/libs/common/logic',
};
