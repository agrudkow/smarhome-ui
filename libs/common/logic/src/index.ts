export * from './lib/use-accordion';
export * from './lib/use-window-dimensions';
