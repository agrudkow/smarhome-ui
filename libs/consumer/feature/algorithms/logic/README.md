# consumer-feature-algorithms-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-algorithms-logic` to execute the unit tests via [Jest](https://jestjs.io).
