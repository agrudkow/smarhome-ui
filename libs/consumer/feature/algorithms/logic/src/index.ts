export * from './lib/algorithm.interface';
export * from './lib/algorithms-cells-parser';
export * from './lib/algorithms-data-parser';
export * from './lib/use-algorithms-list';
export * from './lib/use-algorithm-detials';
export * from './lib/use-algorithm-rating';
export * from './lib/use-select-algorithm';
