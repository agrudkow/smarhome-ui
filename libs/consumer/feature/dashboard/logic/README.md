# consumer-feature-dashboard-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-dashboard-logic` to execute the unit tests via [Jest](https://jestjs.io).
