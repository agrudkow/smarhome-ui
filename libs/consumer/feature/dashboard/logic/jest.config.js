module.exports = {
  name: 'consumer-feature-dashboard-logic',
  preset: '../../../../../jest.config.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'html'],
  coverageDirectory:
    '../../../../../coverage/libs/consumer/feature/dashboard/logic',
};
