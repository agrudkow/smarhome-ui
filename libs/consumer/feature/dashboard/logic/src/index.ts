export * from './lib/use-billings';
