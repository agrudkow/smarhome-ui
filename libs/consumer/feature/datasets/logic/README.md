# consumer-feature-datasets-logic

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test consumer-feature-datasets-logic` to execute the unit tests via [Jest](https://jestjs.io).
