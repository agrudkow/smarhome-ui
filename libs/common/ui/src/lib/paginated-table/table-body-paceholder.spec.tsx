import React from 'react';
import { customRender } from '@smarthome/common/logic';
import TableDataPaceholder from './table-body-paceholder';

describe(' TableDataPaceholder', () => {
  it('should render successfully', () => {
    const { baseElement } = customRender(
      <table>
        <tbody>
          <TableDataPaceholder text="Test text" height={40} colSpan={2} />
        </tbody>
      </table>
    );
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot when expanded', () => {
    const { baseElement } = customRender(
      <table>
        <tbody>
          <TableDataPaceholder text="Test text" height={40} colSpan={2} />
        </tbody>
      </table>
    );
    expect(baseElement.firstChild).toMatchSnapshot();
  });

  it('should set placeholder', () => {
    const { getByText } = customRender(
      <table>
        <tbody>
          <TableDataPaceholder
            text="Test placeholder"
            height={40}
            colSpan={2}
          />
        </tbody>
      </table>
    );
    expect(getByText('Test placeholder')).toBeTruthy();
  });
});
