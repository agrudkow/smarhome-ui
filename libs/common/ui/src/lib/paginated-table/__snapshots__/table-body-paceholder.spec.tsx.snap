// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` TableDataPaceholder should match snapshot when expanded 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  color: #444;
  font-size: 20px;
  font-weight: bold;
  line-height: 20px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 40px;
  opacity: 0.25;
  text-align: center;
}

@media (min-width:1024px) {
  .c1 {
    font-size: 30px;
    -webkit-letter-spacing: -0.3px;
    -moz-letter-spacing: -0.3px;
    -ms-letter-spacing: -0.3px;
    letter-spacing: -0.3px;
    line-height: 30px;
  }
}

<div>
  <table>
    <tbody>
      <tr
        class="MuiTableRow-root"
        style="height: 40px;"
      >
        <td
          class="MuiTableCell-root"
          colspan="2"
        >
          <div
            class="c0"
            height="40"
          >
            <h3
              class="c1"
            >
              Test text
            </h3>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
